h2o-py/h2o/estimators/single_decision_tree.py

```diff
@@ -0,0 +1,151 @@
+        self._id = self._parms['model_id'] = model_id
+        self.training_frame = training_frame
+        self.ignored_columns = ignored_columns
+        self.ignore_const_cols = ignore_const_cols
+        self.categorical_encoding = categorical_encoding
+        self.response_column = response_column
+        self.max_depth = max_depth
+
+    @property
+    def training_frame(self):
+        """
+        Id of the training data frame.
+
+        Type: ``Union[None, str, H2OFrame]``.
+        """
+        return self._parms.get("training_frame")
+
+    @training_frame.setter
+    def training_frame(self, training_frame):
+        self._parms["training_frame"] = H2OFrame._validate(training_frame, 'training_frame')
+
+    @property
+    def ignored_columns(self):
+        """
+        Names of columns to ignore for training.
+
+        Type: ``List[str]``.
+        """
+        return self._parms.get("ignored_columns")
+
+    @ignored_columns.setter
+    def ignored_columns(self, ignored_columns):
+        assert_is_type(ignored_columns, None, [str])
+        self._parms["ignored_columns"] = ignored_columns
+
+    @property
+    def ignore_const_cols(self):
+        """
+        Ignore constant columns.
+
+        Type: ``bool``, defaults to ``True``.
+        """
+        return self._parms.get("ignore_const_cols")
+
+    @ignore_const_cols.setter
+    def ignore_const_cols(self, ignore_const_cols):
+        assert_is_type(ignore_const_cols, None, bool)
+        self._parms["ignore_const_cols"] = ignore_const_cols
+
+    @property
+    def categorical_encoding(self):
+        """
+        Encoding scheme for categorical features
+
+        Type: ``Literal["auto", "enum", "one_hot_internal", "one_hot_explicit", "binary", "eigen", "label_encoder",
+        "sort_by_response", "enum_limited"]``, defaults to ``"auto"``.
+        """
+        return self._parms.get("categorical_encoding")
+
+    @categorical_encoding.setter
+    def categorical_encoding(self, categorical_encoding):
+        assert_is_type(categorical_encoding, None, Enum("auto", "enum", "one_hot_internal", "one_hot_explicit", "binary", "eigen", "label_encoder", "sort_by_response", "enum_limited"))
+        self._parms["categorical_encoding"] = categorical_encoding
+
+    @property
+    def response_column(self):
+        """
+        Response variable column.
+
+        Type: ``str``.
+        """
+        return self._parms.get("response_column")
+
+    @response_column.setter
+    def response_column(self, response_column):
+        assert_is_type(response_column, None, str)
+        self._parms["response_column"] = response_column
+
+    @property
+    def max_depth(self):
+        """
+        Max depth of tree.
+
+        Type: ``int``, defaults to ``20``.
+        """
+        return self._parms.get("max_depth")
+
+    @max_depth.setter
+    def max_depth(self, max_depth):
+        assert_is_type(max_depth, None, int)
+        self._parms["max_depth"] = max_depth
+
```

Comment:
This should not be generated manually.  It should be generated automatically by functions in h2o-bindings.  You need to edit the following files:

gen_python.py
gen_R.py

If you have any specific functions that are specific to single decision trees, you will need to put them in custom/python and custom/R.  You can add a function gen_singledecisiontree.py (or gen_sdt.py or whatever name you like) in python and R directory.  You can take a look at gen_modelselection.py as a reference.  Let me know if you have questions.